extensions/interactions/CodeRepl/CodeClassifierPredictionService.js

```diff
@@ -0,0 +1,217 @@
+// Copyright 2017 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * Code classifier prediction functions.
+ *
+ * IMPORTANT NOTE: The prediction function uses the classifier data
+ * of trained model for inference. These functions must be changed if
+ * there are any changes in corresponding classifier training function on
+ * Oppia-ml.
+ */
+
+oppia.factory('CodeClassifierPredictionService', [
```

Comment:
Maybe call this CodeReplPredictionService so we have a standard naming scheme for these things ("interaction id" + prediction service). Also some changes may be needed to the .py file to say whether such a service exists so that the relevant files can be imported only if necessary?